Add ERC: On-chain NFT royalty enforcement (#224)
ethereum/ERCs: ERCS/erc-7526.md

---
eip: 7526
title: On-chain NFT royalty enforcement

> description: A game-theoretic mechanism to induce truthful revelation of NFT sales and their value to allow for decentralised royalty enforcement.

Reviewer: ```suggestion
description: An interface for game-theoretic mechanism to induce truthful revelation of NFT sales and their value to allow for decentralised royalty enforcement.
```

ERCs are standard track: most ERCs are used to propose a way for interoperability. Using it to propose a "mechanism" is a bit confusing. Maybe add "interface" in your summary.

Also, I don't see game theory being used other than the motivation. Can you show it in the reference implementation of what you refer to?

Royalty systems have therefore relied on external parties to supply this information, which can either introduce centralisation, or be manipulated if the involved parties are dishonest.
This problem is aggravated in over-the-counter (OTC) deals because the flow of funds can be handled entirely via hidden channels.

However, on-chain royalty enforcement can still be feasible in practice by creating a sufficiently strong incentive or penalty to force participating parties to provide honest information.
We describe one such approach, a game-theoretic mechanism that targets the primary actors in a transfer (i.e. the previous and new token owner) as the only players, and induce the new owner to truthfully supply the information required for enforcement.

## Specification

The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.

> For each `tokenId` we introduce a set of addresses whose members MAY transfer the token to themselves without payment by calling `takeBack(uint256 tokenId)`, which we will refer to as a *take-back*.

Reviewer: ```suggestion
For each `tokenId` we introduce a set of addresses whose members MAY transfer the token to themselves without payment by calling `takeBack(uint256 tokenId)`, which we will refer to as a *take-back*.
```
"we introduce a set of addresses": can you ellaborate?

4. The new owner MUST be subject to the same mechanism rules; i.e. take-back, royalty payment, and auto-listing.

If no sale occurs during the auto-listing window, the contract SHOULD transfer the royalty payment to the NFT creator.

Marketplaces facilitating sales SHOULD forward royalties in the same transaction as the sale to guarantee payment within the grace period, thus immediately unfreezing the token.
Marketplaces MAY allow buyers to configure either the auto-listing price or the royalties paid (as the outcome is equivalent), or they MAY provide sensible default values or employ more sophisticated heuristics at their discretion.
If allowing buyers to configure values, marketplaces MAY limit this to the auto-listing price for ease of understanding as royalties are then conceptually equivalent to ERC-2981's `royaltyInfo(autoListingPrice)`, which is a concept familiar to end users.

```solidity
/**
 * @dev This constitutes a rough draft of the minimum required functions, and will almost certainly change as we
 * complete the reference implementation.

>  */

Reviewer 2: Proposals should be written as if they are already complete and ready to use.

If you want to leave yourself notes, you can use HTML-style comments (`<!-- TODO -->`) though they'll have to be outside of the code fence.

interface IERC7526 {
  function takeBack(uint256 tokenId) external;
  function payRoyalty(uint256 tokenId) external payable;
  function buyAutoListing(uint256 tokenId) external payable;
}
```

## Rationale

We have published a formal definition of the mechanism and extensive analyses on arXiv[^1].

[^1]:
    ```csl-json
    {
      "type": "article",
      "id": 1,
      "author": [
        {
          "family": "Huber",
          "given": "David"
        },
        {
          "family": "Schlosberg",
          "given": "Arran"
        }
      ],
      "DOI": "10.48550/arXiv.2403.07896",
      "title": "SACRÉ BLEU: Self-Assessed Creator Royalties Énforced by Balancing Liquidity Estimation & Utility",
      "original-date": {
        "date-parts": [
          [2024, 2, 19]
        ]
      },
      "URL": "https://arxiv.org/abs/2403.07896",
      "custom": {
        "additional-urls": [
        ]
      }
    }

>     ```

Reviewer 2: If you hold the copyrights to the paper, you can instead include it in the `assets/` directory.

> ### Summary rationale for the draft:

Reviewer 2: Again, write as if this was the final version.

* Accepting the risk of take-back in the event of no royalty being paid is only a rational decision if the same entity controls both addresses; i.e. a change in beneficial owner of the token results in a royalty payment but transfer between one's own addresses doesn't require one.
* The auto-listing window ensures *truthful* revelation of the value of the token.

## Backwards Compatibility

No backward compatibility issues found.

## Reference Implementation

> TBD

Reviewer 2: ```suggestion
TBD <!-- TODO -->
```

## Security Considerations

> Needs discussion.

Reviewer 2: ```suggestion
Needs discussion. <!-- TODO -->
```
